exercises/pascals-triangle/example.rs

```diff
@@ -0,0 +1,25 @@
+pub struct PascalsTriangle {
+    row_count: u32,
+}
+
+impl PascalsTriangle {
+    pub fn new(row_count: u32) -> Self {
+        PascalsTriangle { row_count: row_count }
+    }
+
+    pub fn rows(&self) -> Vec<Vec<u32>> {
+        (0..self.row_count).map(|row| PascalsTriangle::row(row)).collect()
+    }
+
+    pub fn row(number: u32) -> Vec<u32> {
+        let mut r = vec![1];
+
+        for p in 1..(number + 1) {
+            if let Some(last) = r.pop() {
+                let next = (last * (number + 1 - p)) / p;
+                r.extend(&[last, next])
```

Comment:
Hmm. I wonder. I thought that was supposed to work, and then it didn't. I quickly tested and found that with the addition of a strategically-placed `&`, it does work:

```rust
    if let Some(&y) = x.last() {
        x.push(y + 5);
    }
```

https://play.rust-lang.org/?gist=a3ace6f29c85b05e9f5027d31d0b4072&version=stable&backtrace=0

I may need a moment to try to puzzle out why one works and not the other. If someone has assistance to offer in this regard, I sure would appreciate it.